api/services/Flairs.js

```diff
@@ -253,3 +278,43 @@ exports.getSimilarBannedFCs = function (fc) {
     });
   });
 };
+
+// Returns a promise of all flair apps for a particular username. If username is undefined, returns flair apps for all users.
+// Note: This will include the claimedBy property, which we probably don't want the user to see.
+exports.getApps = function (username) {
+  var query = username ? {user: username} : {};
+  return Application.find(query).then(function (apps) {
+    apps.forEach(function (app) {
+      app.claimedBy = app_claim_cache.get(app.id) || app_claim_cache.get(app.user);
+    });
+    return apps;
+  });
+};
+
+// Returns a promise for all flairs
+exports.getFlairs = function () {
+  return Flair.find({});
+};
+
+exports.refreshAppClaim = function (ref, mod_username) {
```

Comment:
Whenever a mod clicks a ref link, their client sends a POST request to the server containing info on the ref that they clicked. The server interprets this ref and runs a query to determine which application the mod was working on. If any applications match, they are marked as 'claimed' by the mod, and remain so for 5 minutes.